Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -463,7 +469,15 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
    */
   virtual TYPE &operator=(const TYPE &value) {
     TYPE oldValue = m_value;
-    m_value = value;
+    if (std::is_same<TYPE, std::string>::value) {
```

Comment:
The type here is the type of the ParameterWithValue, not the incoming type of the method.

As far as I know we don't support char* parameters